### DISCUSSION TITLE
Best practice on how to import custom .xarco.urdf

### DISCUSSION THREAD

**Discussion by tombanfi** [foxglove/community#72] [Q&A] [answered]
Hi there,
I was wandering if you may kindly provide some best practice on how to import custom .xarco.urdf file. As an example, I was trying to use studio's URDF viewer to load [this](https://github.com/culurciello/pybullet_ur5_gripper/blob/master/robots/urdf/ur5e.urdf) .xarco.urdf file without success. I tested this using both the .deb image on a Ubuntu 18.04 and using the provided docker image, both test were run using ROS Foxy. A way to get more informative (stack trace?) error messages may also be very helpful to allow independent debug of such issues.

Thank you

**Comment by defunctzombie** [comment, marked as answer]
The error is not very helpful:

<img width="245" alt="image" src="https://user-images.githubusercontent.com/84792/153467076-bd16bf4b-bf50-4fec-b52e-541074504d5e.png">

I looked into this for this file and the reason for the failed to load error is due to the meshes used within the urdf file:

```
<mesh filename="../meshes/ur5e/visual/base.dae"/>
```

Studio tries to open those but it cannot find them since they use relative paths.

For ROS systems, you can use the `package://` prefix and reference the meshes from specific packages. You'll want to set the `ROS_PACKAGE_PATH` setting in Studio accordingly:

https://foxglove.dev/docs/app-concepts/preferences
